src/main/java/org/skriptlang/skript/bukkit/recipes/elements/EffSmithingData.java

```diff
@@ -0,0 +1,69 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.SmithingRecipe;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe.MutableSmithingRecipe;
+import org.skriptlang.skript.bukkit.recipes.CreateRecipeEvent;
+import org.skriptlang.skript.bukkit.recipes.CreateRecipeEvent.SmithingRecipeEvent;
+
+@Name("Recipe Data Component Copy")
+@Description({
+	"Whether or not the data components of the base item should be copied to the result item of the recipe.",
+	"NOTE: This effect only works when creating a Smithing, Smithing Transform, or Smithing Trim Recipe."
+})
+@Examples({
+	"set {_recipe} to a new smithing transform recipe:",
+		"\tset the recipe base item to a raw iron named \"Impure Iron\"",
+		"\tset the recipe additional item to an iron nugget named \"Pure Iron\"",
+		"\tset the recipe result item to an iron ingot named \"Heavenly Iron\"",
+		"\tallow the data component copying from the base item to the result item"
+})
+@RequiredPlugins("Paper 1.20.5+")
+@Since("INSERT VERSION")
+
+public class EffSmithingData extends Effect {
+
+	private static final boolean SUPPORTS_COPY_DATA = Skript.methodExists(SmithingRecipe.class, "willCopyDataComponents");
+
+	static {
+		if (SUPPORTS_COPY_DATA) {
+			Skript.registerEffect(EffSmithingData.class,
+				"allow [the] data component (copying|transfer) from [the] base item to [the] result item",
+				"disallow [the] data component (copying|transfer) from [the] base item to [the] result item");
```

Comment:
```suggestion
				"allow (copying|transfering) [the] data component[s] from [the] base item to [the] result item",
				"disallow (copying|transfering) [the] data component[s] from [the] base item to [the] result item");
```

Reply:
although maybe something like `maintain the data components` would be better, not sure